<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1045</width>
    <height>836</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="2">
     <widget class="QPlainTextEdit" name="console">
      <property name="sizePolicy">
       <sizepolicy hsizetype="MinimumExpanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
    <item row="2" column="1" colspan="2">
     <widget class="QwtPlot" name="qwtPlot"/>
    </item>
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Environment</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <widget class="MinGameWidget" name="gamewidget" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>350</width>
            <height>350</height>
           </size>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Neural Network</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="0">
         <widget class="NNWidget" name="nnwidget" native="true"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>Weights</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="0" column="0">
         <widget class="QTableWidget" name="wtableWidget">
          <column>
           <property name="text">
            <string>in</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>out</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>value</string>
           </property>
          </column>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item row="1" column="0" rowspan="2">
     <widget class="QToolBox" name="toolBox">
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="page">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>508</width>
         <height>268</height>
        </rect>
       </property>
       <attribute name="label">
        <string>Q learning</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="0" column="0">
         <widget class="QPushButton" name="showQ">
          <property name="text">
           <string>Show Q</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QPushButton" name="showOptmalButton">
          <property name="text">
           <string>Show Optimal Trajectory</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QPushButton" name="learnedButton">
          <property name="text">
           <string>Show Agent Trajectory</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QPushButton" name="lernButton">
          <property name="text">
           <string>QLearning</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page_2">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>508</width>
         <height>268</height>
        </rect>
       </property>
       <attribute name="label">
        <string>Page 2</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_6">
        <item row="0" column="0">
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string> Neural Network controls</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_7">
           <item row="0" column="0">
            <layout class="QHBoxLayout" name="horizontalLayout">
             <item>
              <widget class="QPushButton" name="test_nn1">
               <property name="text">
                <string>test_nn1 (learn)</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="pushButton_2">
               <property name="text">
                <string>Stop Learn</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="maxEpochSB">
               <property name="prefix">
                <string>max_epoch =</string>
               </property>
               <property name="maximum">
                <number>100000000</number>
               </property>
               <property name="value">
                <number>100000</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="1" column="0">
            <widget class="QPushButton" name="pushButton">
             <property name="text">
              <string>test_nn2 (evalute sin(x))</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QPushButton" name="resetW">
               <property name="text">
                <string>Reset Weights</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="pushButton_3">
               <property name="text">
                <string>Draw Errors</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="pushButton_4">
               <property name="text">
                <string>Draw W Stat</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="3" column="0">
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1045</width>
     <height>29</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuSave_values">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionSave_values"/>
    <addaction name="actionGenerate_values"/>
    <addaction name="actionSave_NNet_weights"/>
    <addaction name="actionLoad_NNet_weights"/>
    <addaction name="actionClear_NNet"/>
    <addaction name="actionGenerate_NNet"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionShowIDs"/>
   </widget>
   <addaction name="menuSave_values"/>
   <addaction name="menuView"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionSave_values">
   <property name="text">
    <string>Save values</string>
   </property>
  </action>
  <action name="actionGenerate_values">
   <property name="text">
    <string>Generate values</string>
   </property>
  </action>
  <action name="actionShowIDs">
   <property name="text">
    <string>ShowIDs</string>
   </property>
  </action>
  <action name="actionSave_NNet_weights">
   <property name="text">
    <string>Save NNet</string>
   </property>
  </action>
  <action name="actionLoad_NNet_weights">
   <property name="text">
    <string>Load NNet</string>
   </property>
  </action>
  <action name="actionClear_NNet">
   <property name="text">
    <string>Clear NNet</string>
   </property>
  </action>
  <action name="actionGenerate_NNet">
   <property name="text">
    <string>Generate NNet</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QwtPlot</class>
   <extends>QFrame</extends>
   <header>qwt_plot.h</header>
  </customwidget>
  <customwidget>
   <class>MinGameWidget</class>
   <extends>QWidget</extends>
   <header>mingamewidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>NNWidget</class>
   <extends>QWidget</extends>
   <header>nnwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
